File: practices/update_financial_calculator.py
# AB 7th Update Financial Calculator

def get_expense(name, income):
    amount = float(input(f"What is your monthly {name}: "))
    percent = (amount / income) * 100
    print(f"Your {name} is ${amount:.2f} and that is {percent:.0f}% of your income.")
    return amount

def main():
    income = float(input("What is your monthly income: "))

    rent = get_expense("rent/mortgage", income)
    utilities = get_expense("utilities", income)
    groceries = get_expense("groceries", income)
    transportation = get_expense("transportation", income)

    savings = income * 0.10
    print(f"\nYou should save ${savings:.2f} a month, that is 10% of your income.")

    total_expenses = rent + utilities + groceries + transportation + savings
    leftover = income - total_expenses
    print(f"You have ${leftover:.2f} of spending money each month!")

main()

File: c_programming_2/practices/update_financial_calculator.c
// AB Update Financial Calculator

#include <stdio.h>
float stuff(char *info) {
    float value;
    printf("%s", info);
    scanf("%f", &value);
    return value;
}
int print_expense(char *label, float amount, float income) {
    float percent = (amount / income) * 100;
    printf("%s: %.2f which is %.0f percent of your income\n", label, amount, percent);
}
int main() {
    float monthly_income = stuff("What is your monthly income: ");
    float rent = stuff("What is your monthly rent: ");
    float utilities = stuff("What is your monthly utilities cost: ");
    float groceries = stuff("What is your monthly groceries cost: ");
    float transportation = stuff("What is your monthly transportation cost: ");
    float total_expenses = rent + utilities + groceries + transportation;
    float savings = monthly_income * 0.10;
    float spending_money = monthly_income - total_expenses - savings;
    print_expense("Rent", rent, monthly_income);
    print_expense("Utilities", utilities, monthly_income);
    print_expense("Groceries", groceries, monthly_income);
    print_expense("Transportation", transportation, monthly_income);
    float savings_percent = (savings / monthly_income) * 100;
    printf("\nYou should save %.2f a month which is %.0f percent of your income\n", savings, savings_percent);
    printf("You have %.2f of spending money each month\n", spending_money);
    return 0;
}